src/main/java/com/amazon/dlic/auth/http/jwt/keybyoidc/HTTPJwtKeyByOpenIdConnectAuthenticator.java

```diff
@@ -1,73 +0,0 @@
-/*
```

Comment:
@stephen-crawford Is the user-info endpoint only used to extract the backend roles from the external IdP or does it have another use as well? (i.e. custom attributes)

Would it make sense to have a protected method in the AbstractHttpJwtAuthenticator to `extractRoles` and have this class override the implementation?

The class renaming is making it harder to determine what the diffs are in this PR.

Reply:
Got it, I see an example positive response here: https://docs.aws.amazon.com/cognito/latest/developerguide/userinfo-endpoint.html#get-userinfo-response-sample

So userinfo endpoint needs to be called before extractRoles and extractSubject.

Reply:
Yeah, this code is pretty complicated which is why I tried to add a lot of tests. The issue is that the existing OIDC logic was not implemented with the userinfo endpoint in mind. It is tightly coupled to the JWTAuthenticator (and the abstract version), but the userinfo can be a jwt or json which complicates things.

Reply:
@cwperks I added a comment on the test file and the renamed file to help make clear what is new.